src/libraries/System.Formats.Tar/tests/TarWriter/TarWriter.Tests.cs

```diff
@@ -145,6 +145,74 @@ public void Verify_Checksum_SymbolicLink_LongLink(TarEntryFormat format) =>
         public void Verify_Checksum_SymbolicLink_LongLink_LongPath(TarEntryFormat format) =>
             Verify_Checksum_Internal(format, TarEntryType.SymbolicLink, longPath: true, longLink: true);
 
+        [Fact]
+        public void Verify_Size_RegularFile_Empty()
+        {
+            using MemoryStream archiveStream = new();
+            string entryName = "entry.txt";
+            using (TarWriter archive = new(archiveStream, TarEntryFormat.V7, leaveOpen: true))
+            {
+                V7TarEntry e = new(TarEntryType.V7RegularFile, entryName)
+                {
+                    DataStream = new MemoryStream(0)
+                };
+                archive.WriteEntry(e);
+            }
+
+            int sizeLocation = 100 + // Name
+                               8 +   // Mode
+                               8 +   // Uid
+                               8;    // Gid
+            int sizeLength = 12;
+
+            archiveStream.Position = 0;
+            byte[] actual = archiveStream.GetBuffer()[sizeLocation..(sizeLocation + sizeLength)];
+
+            byte[] expected = [0x30, 0x30, 0x30, 0x30, 0x30, 0x30, 0x30, 0x30, 0x30, 0x30, 0x30, 0];
+            AssertExtensions.SequenceEqual(expected, actual);
+
+            archiveStream.Position = 0;
+            using TarReader reader = new(archiveStream);
+
+            TarEntry? actualEntry = reader.GetNextEntry();
+            Assert.NotNull(actualEntry);
+            Assert.Equal(0, actualEntry.Length);
+            Assert.Null(actualEntry.DataStream); // No stream created when size field's value is 0
+        }
+
+        [Fact]
+        public void Verify_Compatibility_RegularFile_EmptyFile_NoSizeStored()
+        {
+            using MemoryStream archiveStream = new();
+            string entryName = "entry.txt";
+            using (TarWriter archive = new(archiveStream, TarEntryFormat.V7, leaveOpen: true))
```

Comment:
Instead of relying on producing this again using the implementation, it might be better to just read from a base64 encoded string so we can be certain that we're testing for the old file format exactly as it was previously created.

Reply:
I'm on the fence if that's more readable or not.  On one hand I'd prefer that compat tests consume inputs that don't change, on the other hand -- this code is more descriptive about what it's doing.

If you want, here's the string, produced by running this and then calling `Convert.ToBase64String`, you could use it like:
```c#
using MemoryStream archiveStream2 = new(Convert.FromBase64String("ZW50cnkudHh0AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAADAwMDA2NDQAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAADE0NzMwMzQ0MjQwADA1MTM2AAAgAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA"));
```